response.go

```diff
@@ -157,6 +170,20 @@ func (r *searchResponse) start(ctx context.Context, searchRequest *SearchRequest
 				case ApplicationSearchResultReference:
 					ref := packet.Children[1].Children[0].Value.(string)
 					r.ch <- &SearchSingleResult{Referral: ref}
+
+				case ApplicationIntermediateResponse:
+					decoded, err := DecodeControl(packet.Children[1])
+					if err != nil {
+						werr := fmt.Errorf("failed to decode intermediate response: %w", err)
+						r.ch <- &SearchSingleResult{Error: werr}
+						return
+					}
+					result := &SearchSingleResult{}
+					result.Controls = append(result.Controls, decoded)
+					r.ch <- result
+
+				default:
+					r.conn.Debug.Printf("got application code: %d", packet.Children[1].Tag)
```

Comment:
Unknown tag shouldn't be treated with a debug message. Please throw an error.

Reply:
Thanks. I fixed it.